schematic/models/validate_attribute.py

```diff
@@ -683,24 +703,22 @@ def regex_validation(
 
         # Validating single re's
         else:
-            manifest_col = manifest_col.astype(str)
             for i, re_to_check in enumerate(manifest_col):
-                if not bool(module_to_call(reg_expression, re_to_check)) and bool(
-                    re_to_check
-                ):
-                    vr_errors, vr_warnings = GenerateError.generate_regex_error(
-                        val_rule=val_rule,
-                        reg_expression=reg_expression,
-                        row_num=str(i + 2),
-                        module_to_call=reg_exp_rules[1],
-                        attribute_name=manifest_col.name,
-                        invalid_entry=manifest_col[i],
-                        dmge=dmge,
-                    )
-                    if vr_errors:
-                        errors.append(vr_errors)
-                    if vr_warnings:
-                        warnings.append(vr_warnings)
+                if not isinstance(re_to_check, NAType):
+                    if not bool(module_to_call(reg_expression, str(re_to_check))):
+                        vr_errors, vr_warnings = GenerateError.generate_regex_error(
+                            val_rule=val_rule,
+                            reg_expression=reg_expression,
+                            row_num=str(i + 2),
+                            module_to_call=reg_exp_rules[1],
+                            attribute_name=manifest_col.name,
+                            invalid_entry=manifest_col[i],
+                            dmge=dmge,
+                        )
+                        if vr_errors:
+                            errors.append(vr_errors)
+                        if vr_warnings:
+                            warnings.append(vr_warnings)
```

Comment:
It's really tough to figure out what's going on here. This is a good candidate for breaking up into several methods. There's too many branches, and the complexity is probably too high.